### PR TITLE
Scheduler Controller Document

### PR DESCRIPTION
**Pull request by CoderKevinZhang**
**What type of PR is this?**
> /kind documentation

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: docs/design-proposals/global-scheduler/GlobalResourceScheduler-SchedulerCrdController.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
+# Global Resource Scheduler – Scheduler CRD
+
+## 1. Module Description
+
+The scheduler module allows the global resource scheduler to add, delete, get, and list existing clusters.
+
```
Comment (cathyhongzhang) [inline docs/design-proposals/global-scheduler/GlobalResourceScheduler-SchedulerCrdController.md:6]: Could you clarify what you mean by "scheduler module"? Suggest to change to:
The scheduler controller will list/watch the API server for scheduler creation/deletion/get/list API calls and then execute the controller logic documented in https://github.com/futurewei-cloud/global-resource-scheduler/blob/master/docs/design-proposals/global-scheduler/GlobalResourceScheduler-ClusterCrdController-ver0.1.md

File: docs/design-proposals/global-scheduler/GlobalResourceScheduler-SchedulerCrdController.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+-   The scheduler runs the scheduling algorithm to select the best cluster for the incoming VM/Container Request.
+
+-   Each scheduler is running as an individual process.
+
+-   Different schedulers are responsible for different clusters.
+
```
Comment (cathyhongzhang) [inline docs/design-proposals/global-scheduler/GlobalResourceScheduler-SchedulerCrdController.md:12]: suggest to change to "each scheduler is responsible for a group/partition of clusters"

File: docs/design-proposals/global-scheduler/GlobalResourceScheduler-SchedulerCrdController.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
+# Global Resource Scheduler – Scheduler CRD
+
+## 1. Module Description
+
+The scheduler module allows the global resource scheduler to add, delete, get, and list existing clusters.
+
+-   The scheduler runs the scheduling algorithm to select the best cluster for the incoming VM/Container Request.
+
+-   Each scheduler is running as an individual process.
+
```
Comment (cathyhongzhang) [inline docs/design-proposals/global-scheduler/GlobalResourceScheduler-SchedulerCrdController.md:10]: HA will be enabled on each active scheduler via endpoint/configmap object lock based leader election mechanism

File: docs/design-proposals/global-scheduler/GlobalResourceScheduler-SchedulerCrdController.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+	// Location represent which geolocation the scheduler is responsible for
+	Location GeolocationInfo `json:"location"`
+
+	// Tag represents the Nth scheduler object
+	Tag string `json:"tag"`
+
+	// Cluster is an array that stores the name of clusters
+	Cluster []string `json:"cluster"`
+}
+
```
Comment (cathyhongzhang) [inline docs/design-proposals/global-scheduler/GlobalResourceScheduler-SchedulerCrdController.md:79]: As discussed in today's meeting, the scheduler controller object should include cluster profile structure consisting of a union of all the clusters' profile attributes.

Comment (cathyhongzhang) [reply docs/design-proposals/global-scheduler/GlobalResourceScheduler-SchedulerCrdController.md:79]: Could you add a structure consisting of union of all the clsuters' profile attributes?

Comment (CoderKevinZhang) [reply docs/design-proposals/global-scheduler/GlobalResourceScheduler-SchedulerCrdController.md:79]: No problem. Thank you for your reminder

**Comment by cathyhongzhang** [comment]
LGTM

**Review by kimeunju108 — APPROVED** [review]
LGTM

**Review by jshaofuturewei — APPROVED** [review]

**Pull request merged**